exchanges/credentials.go

```diff
@@ -185,11 +185,13 @@ func (b *Base) VerifyAPICredentials(creds *account.Credentials) error {
 		return fmt.Errorf("%s %w", b.Name, errRequiresAPIClientID)
 	}
 
-	if b.API.CredentialsValidator.RequiresBase64DecodeSecret && !b.LoadedByConfig {
-		_, err := crypto.Base64Decode(creds.Secret)
+	if b.API.CredentialsValidator.RequiresBase64DecodeSecret && !b.LoadedByConfig && !creds.SecretBase64Decoded {
+		decodedResult, err := crypto.Base64Decode(creds.Secret)
 		if err != nil {
 			return fmt.Errorf("%s API secret %w: %s", b.Name, errBase64DecodeFailure, err)
 		}
+		creds.Secret = string(decodedResult)
+		creds.SecretBase64Decoded = true
 	}
```

Comment:
Suggestion or another PR: Would it be possible to also shift this check. So this gets called when `GetCredentials` is called. I have removed the `b.LoadedByConfig` check in the if statement:
```diff
diff --git a/exchanges/credentials.go b/exchanges/credentials.go
index f8ae323d8..3b79fa9fc 100644
--- a/exchanges/credentials.go
+++ b/exchanges/credentials.go
@@ -83,6 +83,15 @@ func (a *API) SetSubAccount(sub string) {
 // CheckCredentials checks to see if the required fields have been set before
 // sending an authenticated API request
 func (b *Base) CheckCredentials(creds *account.Credentials, isContext bool) error {
+       if b.API.CredentialsValidator.RequiresBase64DecodeSecret && !creds.SecretBase64Decoded {
+               decodedResult, err := crypto.Base64Decode(creds.Secret)
+               if err != nil {
+                       return fmt.Errorf("%s API secret %w: %s", b.Name, errBase64DecodeFailure, err)
+               }
+               creds.Secret = string(decodedResult)
+               creds.SecretBase64Decoded = true
+       }
+
        if b.SkipAuthCheck {
                return nil
        }
@@ -185,14 +194,14 @@ func (b *Base) VerifyAPICredentials(creds *account.Credentials) error {
                return fmt.Errorf("%s %w", b.Name, errRequiresAPIClientID)
        }

-       if b.API.CredentialsValidator.RequiresBase64DecodeSecret && !b.LoadedByConfig && !creds.SecretBase64Decoded {
-               decodedResult, err := crypto.Base64Decode(creds.Secret)
-               if err != nil {
-                       return fmt.Errorf("%s API secret %w: %s", b.Name, errBase64DecodeFailure, err)
-               }
-               creds.Secret = string(decodedResult)
-               creds.SecretBase64Decoded = true
-       }
+       // if b.API.CredentialsValidator.RequiresBase64DecodeSecret && !b.LoadedByConfig && !creds.SecretBase64Decoded {
+       //      decodedResult, err := crypto.Base64Decode(creds.Secret)
+       //      if err != nil {
+       //              return fmt.Errorf("%s API secret %w: %s", b.Name, errBase64DecodeFailure, err)
+       //      }
+       //      creds.Secret = string(decodedResult)
+       //      creds.SecretBase64Decoded = true
+       // }
        return nil
 }
```

So that below passes in the context of Kraken which `RequiresBase64DecodeSecret` without too much fiddle:

```go
superCtx := account.DeployCredentialsToContext(context.Background(), &account.Credentials{
		Key:                 "meow",
		Secret:              "cats===",
		SecretBase64Decoded: false,
	})
```